### ISSUE TITLE
Tear down isn't working 100%

### ISSUE THREAD

**Issue by HakanL** [HakanL/WkHtmlToPdf-DotNet#26]
When running unit test the testhost is kept running which holds a reference to the DLL of WkHtmlToPdf library, so if you're making changes it can't build. It's most likely something that isn't disposed/unloaded properly, but needs investigation.

**Comment by sflusov** [comment]
Hi @HakanL ,

We've occurred with the same problem in integration tests and solved it via adding pdftools to DI container.

Please have a look in our code
```
public void ConfigureServices(IServiceCollection services)
{
    // let's register pdf tools for correct ending the chrome process
    // when application shutting down
    services.AddSingleton(typeof(ITools), typeof(PdfTools);
	
    // this scenario apply only in multi-thread scenario like in a webserver
    // see also: https://github.com/rdvojmoc/DinkToPdf#dependency-injection
    services.AddSingleton(typeof(IConverter), typeof(SynchronizedConverter);
}
```

**Comment by HakanL** [comment]
@sflusov Yeah, that's a workaround (to always hold a reference to it), but the issue is if you want to do a teardown then it's not working.

**Comment by sflusov** [comment]
We have ~1.5k integration tests which use WebApplicationFactory (TestServer) and there is no problem run it again and again.
Maybe you lost dispose  method in the end of test collection.

So If you provide example of code I will try to help you

**Comment by HakanL** [comment]
It's not a problem if you don't try to dispose (which is typically the setup with DI), but this bug was raised because it didn't work to tear it down. We would probably need to add a separate unit test to the project to test repeated setup/teardown, I don't think there is one currently, PRs welcome.

**Comment by HakanL** [comment]
@TomGullen You should always just have a singleton/static instance of the SynchronizedConverter when using this project. I assume the teardown issue has something to do with the native library, but more investigation is needed. The examples in here use the SynchronizedConverter, but feel free to submit PR for how the docs can be improved.

**Comment by HakanL** [comment]
@hidegh Yes, you absolutely have to have the `SynchronizedConverter` as a singleton, but the issue raised here is if you wanted to do a full teardown (not recommended though) then there's something that isn't realized. It shouldn't be an issue in a normal application.
